Added support for delaying audio focus release after finishing playing queued voice instructions.

tomaszrybakiewicz opened the pull request:
Closes [#5264](https://github.com/mapbox/mapbox-navigation-android/issues/5264)

### Description
- Added `VoiceInstructionsPlayerOptions.abandonFocusDelay` option that allows customizing an audio focus release delay.
- Updated `MapboxVoiceInstructionsPlayer` logic to delay audio focus release after playing all queued voice instructions.
- Updated the example app `MapboxVoiceActivity` to test audio focus release delay. (Also added NavigationCamera and recenter button for easier testing).
- Regen Metalava API file for `libnavui-voice` module.
- Minor code cleanup and improvements in `MapboxVoiceInstructionsPlayer` and related classes.

### Testing instructions

1. Play music in a music player app of your choosing.
2. Launch the Example App > Voice Example.
3. Long-press on a map to start navigation. (Hint: Select a destination that will generate a route with maneuvers close together.)
4. Observe the voice instructions player. Notice that it will retain audio focus for the additional 2 seconds after playing instructions and hold the focus if there are multiple instructions to play in a sequence.

VitaminPSG commented:
> @yuryybk @VitaminPSG It's been few days since we requested a review from the team. I just want to make sure that it did not get buried among your other priorities. I would like to request the team to prioritize this and review the PR. We would like to make sure to capture your thoughts on this and if possible run this and verify if this is what you expect.
> 
> /cc @AntonBaikou @zugaldia

I'm working on it

VitaminPSG reviewed (commented):
Can we add a counter or state that the focus is active, now even if the focus is active it requests it again
I've created a test in `tr-retain-focus-test`

tomaszrybakiewicz commented on libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/AudioFocusDelegateProvider.kt:
```diff
@@ -1,23 +1,28 @@
 import android.os.Build
+import com.mapbox.navigation.ui.voice.options.VoiceInstructionsPlayerOptions
 
 internal object AudioFocusDelegateProvider {
```
@abhishek1508 Maybe we should make this and both `PreOreoAudioFocusDelegate`, and `OreoAndLaterAudioFocusDelegate` classes public to allow developers to wrap our AsyncAudioFocusDelegate implementation?

VitaminPSG replied:
Ok, we can add this to `PreOreoAudioFocusDelegate` and `OreoAndLaterAudioFocusDelegate` as classes which interact with AudioManager

tomaszrybakiewicz commented:
@abhishek1508 @Guardiola31337. I've rebased on top of the latest main and added the CHANGELOG entry for the AudioFocusDelegateProvider visibility change.

_NOTE: there are two changelog entries from this PR. One in the "Features" section and one in the "Bug fixes and improvements"._

Guardiola31337 commented:
:shipit:
